Add `eagerness` option description and example for `prefetch()` (#11250)
withastro/docs: src/content/docs/en/guides/prefetch.mdx

prefetch('/about', { ignoreSlowConnection: true });
```

> ### Prefetch strategies with `eagerness` option

Reviewer: ```suggestion
### `eagerness`
```
I think we can just use the option name here for simplicity!

<p><Since v="5.6.0" /></p>

> With [`clientPrerender`](https://docs.astro.build/en/reference/experimental-flags/client-prerender/) experiment enabled, you can use `eagerness` option, following the same API described in [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/speculationrules#eagerness).
> 
> For example, you have a set of links that you want to be prerendered/prefetched but there are too many for browsers to handle (Chrome have some [limits in place](https://developer.chrome.com/blog/speculation-rules-improvements#chrome-limits)).
> To prevent that, you can set `eagerness: 'moderate'` to get advantage of FIFO strategies and browser heuristics to let it decide when prerender/prefetch them and in what order.

Reviewer: ```suggestion
With the experimental [`clientPrerender`](https://docs.astro.build/en/reference/experimental-flags/client-prerender/) flag enabled, you can use the `eagerness` option on `prefetch()` to suggest to the browser how eagerly it should prefetch/prerender link targets.

This follows the same API described in [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/speculationrules#eagerness) and defaults to `immediate` (the most eager option). In decreasing order of eagerness, the other options are `eager`, `moderate`, and `conservative`.

The `eagerness` option allows you to balance the benefit of reduced wait times against bandwidth, memory, and CPU costs for your site visitors. Some browsers, such as Chrome, have [limits in place](https://developer.chrome.com/blog/speculation-rules-improvements#chrome-limits) to guard against over-speculating (prerendering/prefetching too many links).

To use `prefetch()` programmatically with large sets of links, you can set `eagerness: 'moderate'` to get advantage of [First In, First Out (FIFO)](https://en.wikipedia.org/wiki/FIFO_(computing_and_electronics)) strategies and browser heuristics to let the browser decide when to prerender/prefetch them and in what order:
```

> ```astro

Reviewer: ``````suggestion
```astro "{eagerness: 'moderate'}"
``````
I think this will help highlight the option!

prefetch('/about', { ignoreSlowConnection: true });
```

### `eagerness`

> <p><Since v="5.6.0" /></p>

Reviewer: ```suggestion
<p>
**Type:** `'immediate' | 'eager' | 'moderate' | 'conservative'`<br />
**Default:** `'immediate'`<br />
<Since v="5.6.0" />
</p>
```

> With the experimental [`clientPrerender`](]/en/reference/experimental-flags/client-prerender/) flag enabled, you can use the `eagerness` option on `prefetch()` to suggest to the browser how eagerly it should prefetch/prerender link targets.

Reviewer: ```suggestion
With the experimental [`clientPrerender`](/en/reference/experimental-flags/client-prerender/) flag enabled, you can use the `eagerness` option on `prefetch()` to suggest to the browser how eagerly it should prefetch/prerender link targets.
```

> This follows the same API described in [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/speculationrules#eagerness) and defaults to `immediate` (the most eager option). In decreasing order of eagerness, the other options are `eager`, `moderate`, and `conservative`.

Reviewer: ```suggestion
This follows the same API described in the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/speculationrules#eagerness) and defaults to `immediate` (the most eager option). In decreasing order of eagerness, the other options are `eager`, `moderate`, and `conservative`.
```

> The `eagerness` option allows you to balance the benefit of reduced wait times against bandwidth, memory, and CPU costs for your site visitors. Some browsers, such as Chrome, have [limits in place](https://developer.chrome.com/blog/speculation-rules-improvements#chrome-limits) to guard against over-speculating (prerendering/prefetching too many links).

Reviewer: ```suggestion
The `eagerness` option allows you to balance the benefit of reduced wait times against bandwidth, memory, and CPU costs for your site visitors. Some browsers, such as Chrome, have [limits in place to guard against over-speculating](https://developer.chrome.com/blog/speculation-rules-improvements#chrome-limits)  (prerendering/prefetching too many links).
```

> To use `prefetch()` programmatically with large sets of links, you can set `eagerness: 'moderate'` to get advantage of [First In, First Out (FIFO)](https://en.wikipedia.org/wiki/FIFO_(computing_and_electronics)) strategies and browser heuristics to let the browser decide when to prerender/prefetch them and in what order:

Reviewer: ```suggestion
To use `prefetch()` programmatically with large sets of links, you can set `eagerness: 'moderate'` to take advantage of [First In, First Out (FIFO)](https://en.wikipedia.org/wiki/FIFO_(computing_and_electronics)) strategies and browser heuristics to let the browser decide when to prerender/prefetch them and in what order:
```

This follows the same API described in the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/speculationrules#eagerness) and defaults to `immediate` (the most eager option). In decreasing order of eagerness, the other options are `eager`, `moderate`, and `conservative`.

The `eagerness` option allows you to balance the benefit of reduced wait times against bandwidth, memory, and CPU costs for your site visitors. Some browsers, such as Chrome, have [limits in place to guard against over-speculating](https://developer.chrome.com/blog/speculation-rules-improvements#chrome-limits)  (prerendering/prefetching too many links).

 ```astro
---
---
<script>
// Control prefetching eagerness with `experimental.clientPrerender`
import { prefetch } from 'astro:prefetch';

// This page is resource-intensive
prefetch('/data-heavy-dashboard', { eagerness: 'conservative' });

// This page is critical to the visitor's journey
prefetch('/getting-started'); // defaults to `{ eagerness: 'immediate' }`

// This page may not be visited

> prefetch('/our-history', { eagerness: 'moderate' });

Reviewer: ```suggestion
prefetch('/terms-of-service', { eagerness: 'moderate' });
```

Chrome supports `<link rel="prefetch">`. Prefetching works as intended.

> It also fully supports `<script type="speculationrules">` from the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API), which can be used to futher describe [prefetching strategies and rules](#eagerness), enhancing user experience for your Chrome users. You'll need to enable [`clientPrerender`](/en/reference/experimental-flags/client-prerender/) experiment to utilize this functionality with `prefetch()`

Reviewer 2: small typo
```suggestion
It also fully supports `<script type="speculationrules">` from the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API), which can be used to further describe [prefetching strategies and rules](#eagerness), enhancing user experience for your Chrome users. You'll need to enable [`clientPrerender`](/en/reference/experimental-flags/client-prerender/) experiment to utilize this functionality with `prefetch()`
```

> It also fully supports `<script type="speculationrules">` from the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API), which can be used to futher describe [prefetching strategies and rules](#prefetch-strategies-with-eagerness-option), enhancing user experience for your Chrome users. You'll need to enable [`clientPrerender`](https://docs.astro.build/en/reference/experimental-flags/client-prerender/) experiment to utilize this functionality with `prefetch()`

Reviewer: ```suggestion
It also fully supports `<script type="speculationrules">` from the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API), which can be used to futher describe [prefetching strategies and rules](#eagerness), enhancing user experience for your Chrome users. You'll need to enable [`clientPrerender`](/en/reference/experimental-flags/client-prerender/) experiment to utilize this functionality with `prefetch()`
```
